kafka-connect/src/test/java/com/hazelcast/jet/contrib/connect/KafkaConnectRabbitMQIntegrationTest.java

```diff
@@ -0,0 +1,122 @@
+        factory.setHost(rabbitMQContainer.getContainerIpAddress());
+        factory.setPort(rabbitMQContainer.getAmqpPort());
+
+        Connection conn = factory.newConnection();
+        Channel channel = conn.createChannel();
+        channel.queueBind("test-queue", "ex", "test");
+        byte[] messageBodyBytes = "Hello, world!".getBytes(StandardCharsets.UTF_8);
+
+        publish(channel, messageBodyBytes, ITEM_COUNT);
+
+        Properties rabbitMqProperties = new Properties();
+        rabbitMqProperties.setProperty("name", "rabbitmq-source-connector");
+        rabbitMqProperties.setProperty("connector.class", "com.github.jcustenborder" +
+                ".kafka.connect.rabbitmq.RabbitMQSourceConnector");
+        rabbitMqProperties.setProperty("kafka.topic", "messages");
+        rabbitMqProperties.setProperty("rabbitmq.queue", "test-queue");
+        rabbitMqProperties.setProperty("rabbitmq.host", rabbitMQContainer.getContainerIpAddress());
+        rabbitMqProperties.setProperty("rabbitmq.port", rabbitMQContainer.getAmqpPort().toString());
+        rabbitMqProperties.setProperty("rabbitmq.username", rabbitMQContainer.getAdminUsername());
+        rabbitMqProperties.setProperty("rabbitmq.password", rabbitMQContainer.getAdminPassword());
+
+        Pipeline pipeline = Pipeline.create();
+        pipeline.readFrom(KafkaConnectSources.connect(rabbitMqProperties))
+                .withoutTimestamps()
+                .map(record -> Values.convertToString(record.valueSchema(), record.value()))
+                .writeTo(AssertionSinks.assertCollectedEventually(60,
+                        list -> assertEquals(ITEM_COUNT, list.size())));
+
+        JobConfig jobConfig = new JobConfig();
+        jobConfig.addJarsInZip(Objects.requireNonNull(this.getClass()
+                                                          .getClassLoader()
+                                                          .getResource("kafka-connect-rabbitmq-0.0.2-SNAPSHOT.zip"))
+                                      .getPath()
+        );
+
+        Job job = createJetMember().newJob(pipeline, jobConfig);
+
+        sleepAtLeastSeconds(5);
+        publish(channel, messageBodyBytes, 1);
+        channel.close();
+        conn.close();
+
+        try {
+            job.join();
+            fail("Job should have completed with an AssertionCompletedException, but completed normally");
+        } catch (CompletionException e) {
+            String errorMsg = e.getCause().getMessage();
+            assertTrue("Job was expected to complete with AssertionCompletedException, but completed with: "
+                    + e.getCause(), errorMsg.contains(AssertionCompletedException.class.getName()));
+        }
+    }
+
+    private void publish(Channel channel, byte[] messageBodyBytes, int itemCount) throws IOException {
+        for (int i = 0; i < itemCount; i++) {
+            channel.basicPublish("ex", "test", new AMQP.BasicProperties.Builder()
```

Comment:
BasicProperties can be extracted to a constant, no need to build it for each item I guess ?